tools/codegen/dest/register_dispatch_key.py

```diff
@@ -407,14 +436,43 @@ def generate_defn(cpp_sig: CppSignature) -> str:
             # After running meta, op.outputs_ is guaranteed to be valid;
             # add it to the context
             # TODO: handle multi-return
+            assert ConstRefCType(BaseCType("Tensor", structured.out_arguments(self.g)[0].ctype.name)) == \
+                structured.out_arguments(self.g)[0].ctype
             context.append(Expr(
                 expr="op.outputs_[0]",
-                type=structured.out_arguments(self.g)[0].ctype,
+                # TODO: Stop hardcoding that the output type is a Tensor.  Note
+                # that for the codegen here this is fine because outputs_ is
+                # hardcoded to be tensor already
+                type=MutRefCType(BaseCType("Tensor", structured.out_arguments(self.g)[0].ctype.name)),
             ))
 
             # With the expanded context, do the impl call (if not a meta
             # function)
-            if self.dispatch_key != DispatchKey.Meta:
+            if self.dispatch_key == DispatchKey.DefaultBackend:
+                # TODO: https://github.com/pytorch/pytorch/issues/53023
+                out_sig_group = CppSignatureGroup.from_native_function(
+                    self.g.out, method=False, fallback_binding=f.manual_cpp_binding)
+                out_sig = out_sig_group.most_faithful_signature()
+                api_name = out_sig.name()
+                out_exprs = ', '.join(
+                    e.expr for e in translate(
+                        context,
+                        out_sig.arguments(),
+                        method=False
+                    )
+                )
+                # TODO: I think this means structured won't work with method
+                # only functions (but maybe you're saved by faithful? iunno.)
+                # NB: Originally I wrote this as an at::redispatch call, but
+                # I got in trouble because that meant I needed a DispatchKeySet
+                # in the wrapper function, which meant I needed a DispatchKeySet
+                # in the DispatchKeyFunctions declarations, but the defined API
+                # there does NOT permit a dispatch key set.  I think you can
```

Comment:
Hmm I might take a quick look after the PR lands to see what it would take to get the wrapper to take in a DispatchKeySet. Since if we can get these functions to take in that argument, the dispatcher should just plumb the keyset through for free.

Reply:
I briefly attempted it and I mostly concluded that I really wanted DispatchKeySet to be in the argument list, so that translate could handle it automatically.